src/odemis/driver/smaract.py

```diff
@@ -4754,139 +4415,75 @@ def _doReference(self, future):
             logging.exception("Referencing failure.")
             raise
         finally:
-            # If the state of the device is stable and all channels are valid, the referencing procedure
-            # succeeded. Typically, all channels become valid at the same time during the reference procedure.
-            # We still check all of them individually to be sure.
-            if self.IsStable():
-                for ch, num in self._channels.items():
-                    self.referenced._value[ch] = self.IsValid(num)
+            # Update the referenced values for the channels
+            for ch, num in self._channels.items():
```

Comment:
In the original version, it wouldn't claim that the axes are referenced if the system is not stable. Why changing this?
Also, if the system is not stable, don't you want to still raise an exception, as you used to do?

Reply:
Good catch, I made the changes